homework/hw36/hw36.js

```diff
@@ -0,0 +1,261 @@
+    indexedBy[item.name] = item;
+
+    return indexedBy;
+}, {});
+
+console.log(productsById);
+console.log(productsByCategory);
+console.log(productsByName);
+
+// show products of the selected category
+categories.forEach(category => {
+    category.addEventListener('click', () => {
+        products.innerHTML = '';
+        const categoryName = category.textContent;
+        const categoryProductsList = productsByCategory[categoryName];
+        categoryProductsList.forEach(product => {
+            const li = document.createElement('li');
+            const a = document.createElement('a');
+            a.href = '#';
+            a.textContent = product['name'];
+            li.appendChild(a);
+            products.appendChild(li);
+        });
+    });
+});
+
+//const currentProduct 
+
+// show product information on click
+products.addEventListener('click', event => {
+    if (event.target.tagName === 'A') {
+        const productName = event.target.textContent;
+        const productDescription = `This is ${productName}.`;
+        productInfo.querySelector('p').textContent = productDescription;
+        productInfo.classList.add('show');
+        orderBtn.style.display = "block";
+        orderConfirmation.style.display = "none";
+    }
+});
+
+
+const orderBtn = document.querySelector('.order-btn');
+const orderForm = document.querySelector("#order-form");
+
+orderBtn.addEventListener("click", showOrderForm);
+
+function showOrderForm() {
+    orderBtn.style.display = "none";
+    orderForm.style.display = "block";
+}
+
+const buyBtn = document.querySelector("#buy-btn");
+buyBtn.addEventListener("click", submitOrder);
+
+const orderConfirmation = document.querySelector("#order-confirmation")
+
+function submitOrder(event) {
+    event.preventDefault();
+    if (validateForm()) {
+        orderForm.style.display = "none";
+        orderConfirmation.style.display = "block";
+        displayOrderConfirmation();
+    }
+}
+
+function validateForm() {
+    const name = document.getElementById("name-input").value.trim();
+    const city = document.getElementById("city-select").value.trim();
+    const postOffice = document.getElementById("post-office-input").value.trim();
+    const payment = document.querySelector('input[name="payment"]:checked');
+    const quantity = document.getElementById("quantity-input").value.trim();
+
+    if (!name || !city || !postOffice || !payment || !quantity) {
+        const errorDiv = document.createElement("div");
+        errorDiv.style.color = "red";
+        errorDiv.textContent = "Please fill in all required fields.";
+        document.querySelector("form").appendChild(errorDiv);
+        return false;
+    }
+
+    return true;
+}
+
+function displayOrderConfirmation() {
+    const p = document.getElementById("product-name");
+    const productName = p.innerHTML.slice(8, -1);
+    const product = productsByName[productName];
+
+    const name = document.getElementById("name-input").value.trim();
```

Comment:
A good practice before sending data

Also, you can write a common function for it

```
function normalize(rawValue: any) {
    if (rawValue == null) {
        return null;
    }

    const value = (
        typeof rawValue === 'string'
            ? rawValue.trim()
            : rawValue
    );

    return value === '' ? null : value;
}

// ...

const name = normalize(document.getElementById("name-input").value);
```